src/parachain/rewards.ts

```diff
@@ -9,244 +10,384 @@ import {
      * @returns The reward per token, as a Big object
      */
     getRewardsPoolRewardPerToken(currencyId: CurrencyIdLiteral): Promise<Big>;
     /**
      * Compute the total reward, including the staking (local) pool and the rewards (global) pool
-     * @param vaultId The account ID of the staking pool nominee
+     * @param vaultAccountId The vault ID whose reward pool to check
      * @param nominatorId The account ID of the staking pool nominator
+     * @param vaultCollateralIdLiteral Collateral used by the vault
+     * @param rewardCurrencyIdLiteral The reward currency
      * @returns A Monetary.js amount object, representing the total reward in the given currency
      */
-    computeReward(vaultId: string, nominatorId: string): Promise<MonetaryAmount<Currency<BitcoinUnit>, BitcoinUnit>>;
+    computeReward(
+        vaultAccountId: AccountId,
+        nominatorId: AccountId,
+        vaultCollateralIdLiteral: CollateralIdLiteral,
+        rewardCurrencyIdLiteral: WrappedIdLiteral
+    ): Promise<MonetaryAmount<Currency<BitcoinUnit>, BitcoinUnit>>;
     /**
-     * @param vaultId The vault account ID
+     * @param vaultAccountId The vault ID whose reward pool to check
+     * @param vaultCollateralIdLiteral Collateral used by the vault
+     * @param rewardCurrencyIdLiteral The reward currency
      * @returns The total wrapped token reward collected by the vault
      */
-    getFeesWrapped(vaultId: string): Promise<MonetaryAmount<WrappedCurrency, BitcoinUnit>>;
+    getFeesWrapped(
+        vaultAccountId: AccountId,
+        vaultCollateralIdLiteral: CollateralIdLiteral,
+        rewardCurrencyIdLiteral: WrappedIdLiteral
+    ): Promise<MonetaryAmount<Currency<BitcoinUnit>, BitcoinUnit>>;
 }
 
 export class DefaultRewardsAPI implements RewardsAPI {
     constructor(
         public api: ApiPromise,
         private btcNetwork: Network,
         private electrsAPI: ElectrsAPI,
-        private wrappedCurrency: WrappedCurrency
+        private wrappedCurrency: WrappedCurrency,
+        private nativeCurrency: CollateralCurrency
     ) {}
 
     async computeRewardInStakingPool(
-        vaultId: string,
-        nominatorId: string
+        vaultAccountId: AccountId,
+        nominatorId: AccountId,
+        collateralCurrencyId: CollateralIdLiteral,
+        nonce?: number
     ): Promise<MonetaryAmount<Currency<BitcoinUnit>, BitcoinUnit>> {
-        const currencyId = tickerToCurrencyIdLiteral(this.wrappedCurrency.ticker);
+        const wrappedCurrencyId = tickerToCurrencyIdLiteral(this.wrappedCurrency.ticker) as WrappedIdLiteral;
         const [stake, rewardPerToken, rewardTally] = await Promise.all([
-            this.getStakingPoolStake(currencyId, vaultId, nominatorId),
-            this.getStakingPoolRewardPerToken(currencyId, vaultId),
-            this.getStakingPoolRewardTally(currencyId, vaultId, nominatorId),
+            this.getStakingPoolStake(collateralCurrencyId, vaultAccountId, nominatorId, nonce),
```

Comment:
This is what I mean with consistent ordering, it's a bit confusing when the order changes between different calls.